src/main/java/com/github/manolo8/darkbot/extensions/plugins/PluginHandler.java

```diff
@@ -187,9 +193,19 @@ private void loadPlugins(File[] pluginFiles, List<Plugin> previousPlugins) {
             }
         }
         PLUGIN_CLASS_LOADER = new PluginClassLoader(LOADED_PLUGINS.stream().map(Plugin::getJar).toArray(URL[]::new));
+        pluginEntryNames.entrySet().stream()
+                .filter(plugin -> LOADED_PLUGINS.contains(plugin.getKey()))
+                .forEach(plugin ->
+                        plugin.getKey().getDefinition().features = plugin.getValue().stream()
+                        .filter(file -> file.endsWith(".class"))
+                        .map(class_ -> class_.replace("/", ".").replace(".class", ""))
+                        .filter(class_ -> plugin.getKey().getBasePackage() == null || class_.startsWith(plugin.getKey().getBasePackage()))
+                        .filter(this::isFeature)
+                        .toArray(String[]::new)
+                );
```

Comment:
```suggestion
        pluginClasses.forEach((plDef, cls) ->
                        plDef.features = cls.stream().filter(this::isFeature).toArray(String[]::new));
```
This should be significantly easier with the other changes